### PR TITLE
Test and build the Trin workspace in `win-build` CI pipeline

### PR DESCRIPTION
**Pull request by ogenev**
### What was wrong?
Since genesis, win-build workflow in CI doesn't run all tests in the workspace.

### How was it fixed?
- Add `--workspace` flag to `cargo build` and `cargo test`
- Update the` clippy` command to check for all features and remove depreciated `--all` flag.
-  Add windows support for `trin-cli`. (this issue was found with the updated clippy command)

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] Add entry to the [release notes](https://github.com/ethereum/trin/blob/master/newsfragments/README.md) (may forgo for trivial changes)
- [ ] Clean up commit history

### CODE REVIEW

**Comment by ogenev** [comment]
Should we merge this PR first and then ignore/temporarily stop the win pipeline until we fix all failing tests?

**Comment by njgheorghita** [comment]
Nice catch! I'm of the opinion that it'd be better to bump the priority of this pr and fix failing tests here before merging. Any reason that you'd want to merge this with a failing `win-build`?

**Comment by mrferris** [comment]
I agree with fixing failing tests here, as long we can in a reasonable amount of time. Otherwise I can see it being better to merge this (so that other PRs can make use of the genius idea of running unit tests on windows) and then fix the broken tests in a separate PR.

Anyway, the failures are all from one issue: the sqlite interface seems to want to have sole ownership of the entire `C:\Users\circleci\AppData\Local\Temp\trin\.tmp` directory instead of just the `\trin_abcdef\trin.sqlite` subdirectory of that directory that we're giving it. Notable that this only happens when using a temp directory, and not with the default data directory when running, for example, the peertest. I'm not yet quite seeing why that's happening. Any ideas @njgheorghita?

**Comment by njgheorghita** [comment]
Hmmm... haven't pinned it down just yet, but I imagine that it's related to [this](https://docs.rs/tempfile/latest/tempfile/struct.TempDir.html#resource-leaking) and our use of `TempDir` in `trin_core::utils::db::setup_temp_dir` in those tests that are failing

**Comment by mrferris** [comment]
I was thinking the same, but the fact that the peertest succeeds even though it also uses `setup_temp_dir` (because it uses the `--ephemeral` flag) makes me think that it's not related.

Do we have a reason for using such an old verison of the windows circle ci orb? We're using `2.1` and `5.0` is out. The unit tests should still pass on `2.1` (which uses windows server 2019) though, so I don't think that's related either.

**Comment by mrferris** [comment]
@ogenev if you want to enable "allow edits by maintainers" on this PR I'll push the commit that fixes the unit tests to this PR, or we can just merge this and then I'll merge [this PR](https://github.com/ethereum/trin/pull/404) where I tested the fix

**Comment by mrferris** [comment]
@ogenev alternatively, we can just merge #404 and close this. I'll open that one for review/merge unless we agree otherwise.

**Review by carver — APPROVED** [review]
Looks good. At this point, it's probably faster to just merge #404 which has imported all these commits, and fixed the windows tests.

**Comment by ogenev** [comment]
Closing this in favor of #404.

**Pull request closed without merging**
